src/http/client.cc

```diff
@@ -141,8 +141,17 @@ future<connection::reply_ptr> connection::recv_reply() {
                 throw std::system_error(ECONNABORTED, std::system_category());
             }
             if (parser.failed()) {
-                http_log.trace("Parsing response failed");
-                throw std::runtime_error("Invalid http server response");
+                http_log.warn("Parsing response failed.{}", http_log.level() == log_level::debug ? format(" Reason: {}", parser.error_message()) : "");
```

Comment:
Not sure if it's nice to spam logs with all server-related errors that are going to be "workarounded" anyway. If applying [this](https://github.com/scylladb/seastar/pull/2731/commits/8eea143389d341ac5ddbd0e3cb1cd5bf2c8736ba#r2035044584) suggestion, the exact parsing error can be printed by the caller in case retry is not performed.

Reply:
then just 
```
http_log.trace("Parsing response failed");
try {
        throw std::system_error(EPIPE, std::system_category());
} catch (...) {
        std::throw_with_nested(std::runtime_error(format("Invalid http server response. Reason: {}", parser.error_message())));
}
```
will suffice I guess, lemme check

Reply:
how about now?